### PR TITLE
PWN-7625 - Add Ethereum bundle signature

### PR DESCRIPTION
**Pull request by Davran331122**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7625

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/bridge/claim/ui/ClaimPresenter.kt [hunk 1]
```diff
@@ -154,7 +158,16 @@ class ClaimPresenter(
     }
 
     override fun onSendButtonClicked() {
-        // TODO implement claim logic
+        launch {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/bridge/claim/ui/ClaimPresenter.kt:161]: if(latestBundle == null) return 

so no need to launch at all here
or make it with let
but do not launch and then check

File: app/src/main/java/org/p2p/wallet/bridge/claim/ui/ClaimPresenter.kt [hunk 2]
```diff
@@ -130,20 +134,20 @@ class ClaimPresenter(
...
+    private fun BridgeBundleFee?.toBridgeAmount(
         tokenSymbol: String,
-        decimals: Int
+        decimals: Int,
     ): BridgeAmount {
         return BridgeAmount(
             tokenSymbol = tokenSymbol,
-            tokenAmount = amountInToken(decimals).takeIf { it.isNotZero() },
-            fiatAmount = amountInUsd.takeIf { it.isNotZero() }
+            tokenAmount = this?.amountInToken(decimals).takeIf { !it.isNullOrZero() },
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/bridge/claim/ui/ClaimPresenter.kt:149]: .takeIf { in this case in odd

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -28,6 +28,7 @@
         android:dataExtractionRules="@xml/data_backup_rules"
         android:fullBackupContent="@xml/backup_rules"
         android:icon="@mipmap/ic_launcher"
+        android:largeHeap="true"
```
Comment (gslevinkov) [inline app/src/main/AndroidManifest.xml:31]: revert
again.

File: app/src/main/java/org/p2p/wallet/bridge/api/request/SendEthereumBundleRpcRequest.kt [hunk 4]
```diff
@@ -7,7 +7,7 @@ import org.p2p.wallet.bridge.model.BridgeBundle
 data class SendEthereumBundleRpcRequest(
     @Transient val bundleRequest: BridgeBundle,
 ) : JsonRpc<Map<String, Any>, Unit>(
-    method = "send_ethereum_bundle",
+    method = "send_ethereum_bundle", // "send_ethereum_bundle",
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/api/request/SendEthereumBundleRpcRequest.kt:10]: why comment?

File: app/src/main/java/org/p2p/wallet/bridge/api/response/BridgeBundleFeeResponse.kt [hunk 5]
```diff
@@ -1,11 +1,14 @@
...
     @SerializedName("amount")
-    val amount: BigDecimal,
+    val amount: String,
     @SerializedName("usd_amount")
-    val usdAmount: BigDecimal
+    val usdAmount: String,
+    @SerializedName("chain")
+    val chain: String,
+    @SerializedName("token")
+    val token: String
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/api/response/BridgeBundleFeeResponse.kt:13]: tokenName / tokenSymbol?

File: app/src/main/java/org/p2p/wallet/bridge/claim/ui/ClaimPresenter.kt [hunk 6]
```diff
@@ -154,7 +162,43 @@ class ClaimPresenter(
...
+                    date = transactionDate,
+                    tokenUrl = tokenToClaim.iconUrl.orEmpty(),
+                    amountTokens = amountTokens,
+                    amountUsd = amountUsd.asNegativeUsdTransaction(),
+                    recipient = null,
+                    totalFees = feeList.mapNotNull { it.toTextHighlighting() }
+                )
+                view?.showProgressDialog(lastBundle.bundleId, progressDetails)
+            } catch (e: BridgeResult.Error) {
+                Timber.e(e)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/claim/ui/ClaimPresenter.kt:189]: add some comment after e
Timber.e(e, "Failed to signature") or some stuff

File: ethereumkit/src/main/java/org/p2p/ethereumkit/external/repository/EthereumRepository.kt [hunk 7]
```diff
@@ -2,11 +2,16 @@ package org.p2p.ethereumkit.external.repository
...
 import org.p2p.core.wrapper.eth.EthAddress
+import org.p2p.ethereumkit.internal.models.RawTransaction
+import org.p2p.ethereumkit.internal.models.Signature
 
 interface EthereumRepository {
     fun init(seedPhrase: List<String>)
     suspend fun getBalance(): BigInteger
     suspend fun loadWalletTokens(): List<Token.Eth>
     suspend fun getAddress(): EthAddress
+    fun getPrivateKey(): BigInteger
```
Comment (gslevinkov) [inline ethereumkit/src/main/java/org/p2p/ethereumkit/external/repository/EthereumRepository.kt:15]: which private key are you getting here?

File: ethereumkit/src/main/java/org/p2p/ethereumkit/external/repository/EthereumKitRepository.kt [hunk 8]
```diff
@@ -37,6 +43,21 @@ internal class EthereumKitRepository(
         )
     }
 
+    override fun getPrivateKey(): BigInteger {
+        return tokenKeyProvider?.privateKey ?: throwInitError()
+    }
+
+    override fun signTransaction(transaction: HexString): Pair<Signature, String> {
```
Comment (gslevinkov) [inline ethereumkit/src/main/java/org/p2p/ethereumkit/external/repository/EthereumKitRepository.kt:50]: Pair<Signature, HexString>

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/bridge/api/request/SendEthereumBundleRpcRequest.kt (hunk 4 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/bridge/api/request/SendEthereumBundleRpcRequest.kt:10]: will be switched on the release method on release

**Review by gslevinkov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

File: ethereumkit/src/main/java/org/p2p/ethereumkit/external/repository/EthereumRepository.kt (hunk 7 above)
Comment (Davran331122) [reply ethereumkit/src/main/java/org/p2p/ethereumkit/external/repository/EthereumRepository.kt:15]: It's Ethereum Repository, so it's eth private key )

File: ethereumkit/src/main/java/org/p2p/ethereumkit/external/repository/EthereumKitRepository.kt (hunk 8 above)
Comment (Davran331122) [reply ethereumkit/src/main/java/org/p2p/ethereumkit/external/repository/EthereumKitRepository.kt:50]: I revert it ti Signature

**Pull request merged**